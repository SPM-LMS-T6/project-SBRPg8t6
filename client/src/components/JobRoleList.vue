<template>
  <div>
    <!-- Conditional rendering for when no job roles are available -->
    <div v-if="jobRoles.length === 0">
      <p>No job roles available.</p>
    </div>

    <!-- Job role list -->
    <ul v-else>
      <li v-for="(jobRole, index) in jobRoles" :key="index" class="job-role-item">
        {{ jobRole.role_listing_id }}
        {{ jobRole.role_id }}
        {{ jobRole.role_listing_desc }}
        {{ jobRole.role_listing_source }}
        {{ jobRole.role_listing_open }}
        {{ jobRole.role_listing_close }}
        {{ jobRole.role_description }}
        {{ jobRole.role_name }}
        {{ jobRole.role_status }}
        {{ jobRole.role_skills }}
        <!-- You can display additional job role information here -->
      </li>
    </ul>
  </div>
</template>

<script setup>
import { ref, onMounted } from 'vue'
const jobRoles = ref([])

onMounted(() => {
  jobRoles.value = [
    {
      role_listing_id: 1,
      role_id: 101,
      role_listing_desc: 'This is a sample job role description.',
      role_listing_source: 201,
      role_listing_open: '2023-08-01',
      role_listing_close: '2023-08-15',
      role_description: 'Learning Facilitator',
      role_name: 'Talent Attraction',
      role_status: 'active',
      role_skills: ['Communication', 'Training', 'Problem Solving']
    },
    {
      role_listing_id: 2,
      role_id: 102,
      role_listing_desc: 'Join our dynamic team and contribute to our success.',
      role_listing_source: 202,
      role_listing_open: '2023-08-10',
      role_listing_close: '2023-08-24',
      role_description: 'Software Engineer',
      role_name: 'Engineering',
      role_status: 'active',
      role_skills: ['JavaScript', 'React', 'Node.js']
    },
    {
      role_listing_id: 3,
      role_id: 103,
      role_listing_desc: 'Exciting opportunity for a creative mind.',
      role_listing_source: 203,
      role_listing_open: '2023-07-25',
      role_listing_close: '2023-08-08',
      role_description: 'Graphic Designer',
      role_name: 'Creative Team',
      role_status: 'inactive',
      role_skills: ['Graphic Design', 'Adobe Creative Suite', 'Illustration']
    },
    {
      role_listing_id: 4,
      role_id: 104,
      role_listing_desc: 'Lead our marketing efforts and make an impact.',
      role_listing_source: 204,
      role_listing_open: '2023-08-05',
      role_listing_close: '2023-08-19',
      role_description: 'Marketing Manager',
      role_name: 'Marketing',
      role_status: 'active',
      role_skills: ['Marketing Strategy', 'Digital Marketing', 'Analytics']
    },
    {
      role_listing_id: 5,
      role_id: 105,
      role_listing_desc: 'Join our finance team and drive financial excellence.',
      role_listing_source: 205,
      role_listing_open: '2023-08-15',
      role_listing_close: '2023-08-29',
      role_description: 'Financial Analyst',
      role_name: 'Finance',
      role_status: 'active',
      role_skills: ['Financial Analysis', 'Budgeting', 'Excel']
    }
  ]
})
</script>

<style scoped>
/* Add your component-specific styles here */
.job-role-item {
  cursor: pointer;
  /* Add any other styling you need */
}
</style>
