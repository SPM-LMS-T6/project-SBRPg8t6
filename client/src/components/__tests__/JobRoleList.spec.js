import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import JobRoleList from '../JobRoleList.vue'

describe('JobRoleList', () => {
  it('displays "No job roles available" message when no job roles are provided', () => {
    const jobRoles = []
    const wrapper = mount(JobRoleList, {
      props: { jobRoles }
    })

    expect(wrapper.text()).toContain('No job roles available.')
  })

  it('displays job roles when job roles are provided', async () => {
    const jobRoles = [
      {
        role_listing_id: 1,
        role_id: 101,
        role_listing_desc: 'This is a sample job role description.',
        role_listing_source: 201,
        role_listing_open: '2023-08-01',
        role_listing_close: '2023-08-15',
        role_description: 'Learning Facilitator',
        role_name: 'Talent Attraction',
        role_status: 'active',
        role_skills: ['Communication', 'Training', 'Problem Solving']
      }
      // Add other job role objects here...
    ]

    const wrapper = mount(JobRoleList, {
      props: { jobRoles }
    })

    await wrapper.vm.$nextTick()

    // You can add more specific assertions here to check for job roles.
    // For example: expect(wrapper.text()).toContain('Learning Facilitator');
  })

  it('does not display job roles when no job roles are provided', async () => {
    const jobRoles = []
    const wrapper = mount(JobRoleList, {
      props: { jobRoles }
    })

    await wrapper.vm.$nextTick()

    // Add specific assertions to check that job roles are not displayed.
    // For example: expect(wrapper.text()).not.toContain('Learning Facilitator');
  })

  it('displays job roles for HR_Admin user type', async () => {
    const jobRoles = [
      {
        role_listing_id: 1,
        role_id: 101,
        role_listing_desc: 'This is a sample job role description.',
        role_listing_source: 201,
        role_listing_open: '2023-08-01',
        role_listing_close: '2023-08-15',
        role_description: 'Learning Facilitator',
        role_name: 'Talent Attraction',
        role_status: 'active',
        role_skills: ['Communication', 'Training', 'Problem Solving']
      }
      // Add other job role objects here...
    ]

    const wrapper = mount(JobRoleList, {
      props: { jobRoles }
    })

    await wrapper.vm.$nextTick()

    // Add specific assertions here to check that job roles are displayed for HR_Admin user type.
    // For example: expect(wrapper.text()).toContain('Learning Facilitator');
  })
})
