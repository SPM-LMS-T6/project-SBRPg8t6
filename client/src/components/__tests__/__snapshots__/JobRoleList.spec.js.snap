// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JobRoleList > matches snapshot 1`] = `
"<div data-v-5e29d59b=\\"\\">
  <!-- Conditional rendering for when no job roles are available -->
  <!-- Job role list -->
  <ul data-v-5e29d59b=\\"\\">
    <li data-v-5e29d59b=\\"\\" class=\\"job-role-item\\">Software Engineer
      <!-- You can display additional job role information here -->
    </li>
    <li data-v-5e29d59b=\\"\\" class=\\"job-role-item\\">Product Manager
      <!-- You can display additional job role information here -->
    </li>
  </ul>
</div>"
`;
